flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,281 @@
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    def get(self, entry):
+        return self._find_entry(entry)
+
+    def generate_series_id(self, season, episode=None):
+        tempid = 'S' + str(season.get('seasonNumber')).zfill(2)
+        if episode:
+            tempid = tempid + 'E' + str(episode.get('episodeNumber')).zfill(2)
+
+        return tempid
+
+    def generate_title(self, item, season=None, episode=None):
+        temptitle = item.get('title')
+
+        if item.get('releaseDate') and self.config.get('include_year'):
+            temptitle = temptitle + ' (' + str(item.get('releaseDate')[0:4]) +')'
+
+        if season or episode:
+            temptitle = temptitle + ' ' + self.generate_series_id(season, episode)
+            if episode:
+                if episode.get('title') and self.config.get('include_ep_title'):
+                    temptitle = temptitle + ' ' + episode.get('title')
+
+        return temptitle
+
+    def get_access_token(self):
+        parsedurl = urlparse(self.config.get('base_url'))
+        url = '%s://%s:%s%s/api/v1/Token' % (parsedurl.scheme, parsedurl.netloc, self.config.get('port'), parsedurl.path)
+        data = {'username': self.config.get('username'),
+                'password': self.config.get('password')
+        }
+        headers = { 'Content-Type': 'application/json',
+                    'Accept': 'application/json'
+        }
+        try:
+            log.debug('Logging in with username and password to get access token')
+            log.debug('URL %s', url)
+            access_token = requests.post(url, json=data, headers=headers).json().get('access_token')
+            return access_token
+        except RequestException as e:
+            raise plugin.PluginError('Ombi username and password login failed: %s' % e)
+
+    def ombi_auth(self):
+        if self.config.get('api_key'):
+            log.debug('api_key auth selected')
+            api_key = self.config.get('api_key')
+            header = {'apikey': api_key}
+            return header
+        elif self.config.get('username') and self.config.get('password'):
+            log.debug('username and password auth selected')
+            access_token = self.get_access_token()
+            return {"Authorization": "Bearer %s" %access_token}
+        else:
+            raise plugin.PluginError('Error: an api_key or username and password must be configured')
+
+    def get_request_list(self):
+        auth_header = self.ombi_auth()
+
+        parsedurl = urlparse(self.config.get('base_url'))
+        url=''
+        if self.config.get('type') in ['movies']:
+            log.debug('Received movie list request')
+            url = '%s://%s:%s%s/api/v1/Request/movie' % (parsedurl.scheme, parsedurl.netloc, self.config.get('port'), parsedurl.path)
+        elif self.config.get('type') in ['shows', 'seasons', 'episodes']:
+            log.debug('Received TV list request')
```

Comment:
I don't think these log lines are necessary. If we need to debug it, we always ask for a user's config.